pkg/user/reaper/user.go

```diff
@@ -0,0 +1,136 @@
+package reaper
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/labels"
+
+	"github.com/openshift/origin/pkg/client"
+)
+
+func NewUserReaper(
+	userClient client.UsersInterface,
+	groupClient client.GroupsInterface,
+	clusterBindingClient client.ClusterRoleBindingsInterface,
+	bindingClient client.RoleBindingsNamespacer,
+	sccClient kclient.SecurityContextConstraintsInterface,
+) kubectl.Reaper {
+	return &UserReaper{
+		userClient:           userClient,
+		groupClient:          groupClient,
+		clusterBindingClient: clusterBindingClient,
+		bindingClient:        bindingClient,
+		sccClient:            sccClient,
+	}
+}
+
+type UserReaper struct {
+	userClient           client.UsersInterface
+	groupClient          client.GroupsInterface
+	clusterBindingClient client.ClusterRoleBindingsInterface
+	bindingClient        client.RoleBindingsNamespacer
+	sccClient            kclient.SecurityContextConstraintsInterface
+}
+
+// Stop on a reaper is actually used for deletion.  In this case, we'll delete referencing identities, clusterBindings, and bindings,
+// then delete the user
+func (r *UserReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) (string, error) {
+	removedSubject := kapi.ObjectReference{Kind: "User", Name: name}
+
+	// Remove user from cluster-level role bindings
+	clusterBindings, err := r.clusterBindingClient.ClusterRoleBindings().List(labels.Everything(), fields.Everything())
```

Comment:
Isn't this code completely common to the `GroupReaper` above?  Pass in a subject and a `CusterRoleBindingsInterface`?  Seems like `RoleBindings` would also be congruent and would de-dup the implementation